x-pack/plugins/observability_solution/infra/server/lib/alerting/inventory_metric_threshold/inventory_metric_threshold_executor.ts

```diff
@@ -109,10 +103,6 @@ export const createInventoryMetricThresholdExecutor =
 
     const { share } = libs.plugins;
     const alertsLocator = share.setup.url.locators.get<AlertsLocatorParams>(alertsLocatorID);
```

Comment:
Should we pass the `alertsLocator` in `InfraLocators`?

Reply:
We could, and thus clean up this sort of thing, though there are other parts that would benefit from the same treatment. So perhaps we just box the scope to cover the current changed executors? Anything else should be part of a separate PR to refactor the locator injection